docroot/modules/custom/va_gov_post_api/src/Service/PostFacilityService.php

```diff
@@ -105,23 +113,24 @@ class PostFacilityService extends PostFacilityBase {
    *   The PostAPI service.
    * @param \Drupal\Core\Render\Renderer $renderer
    *   Core renderer.
-   * @param Drupal\Core\File\FileUrlGeneratorInterface $file_url_generator
-   *   File URL generator.
+   * @param \Drupal\Core\File\FileSystemInterface $file_system
+   *   File system service.
+   * @param \Drupal\file\FileRepositoryInterface $file_repository
+   *   File repository service.
    */
-  public function __construct(ConfigFactoryInterface $config_factory, EntityTypeManagerInterface $entity_type_manager, LoggerChannelFactoryInterface $logger_channel_factory, MessengerInterface $messenger, AddToQueue $post_queue, Renderer $renderer, FileUrlGeneratorInterface $file_url_generator) {
+  public function __construct(ConfigFactoryInterface $config_factory, EntityTypeManagerInterface $entity_type_manager, LoggerChannelFactoryInterface $logger_channel_factory, MessengerInterface $messenger, AddToQueue $post_queue, Renderer $renderer, FileSystemInterface $file_system, FileRepositoryInterface $file_repository) {
     parent::__construct($config_factory, $entity_type_manager, $logger_channel_factory, $messenger, $post_queue);
     $this->renderer = $renderer;
-
-    // Set the logging file path.
-    $file_url = $file_url_generator->generateAbsoluteString("public://post_api_force_queue");
-    $log_path = parse_url($file_url, PHP_URL_PATH);
-    // Strip leading slash.
-    $log_path = substr($log_path, 1);
-    if (!is_dir($log_path)) {
-      mkdir($log_path);
+    $this->fileRepository = $file_repository;
+    $this->fileSystem = $file_system;
+    try {
+      self::$logFile = $this->createLogFile($this->fileRepository);
+    }
+    catch (\Exception $e) {
+      $message = sprintf('VA.gov Post API: Failed to create log file. %e', $e->getMessage());
```

Comment:
```suggestion
      $message = sprintf('VA.gov Post API: Failed to create log file. %s', $e->getMessage());
```
![image](https://github.com/swirtSJW/va.gov-cms/assets/5752113/18b9566d-ff10-43df-b85d-04ec422b9835)